src/String.cc

```diff
@@ -228,7 +228,7 @@ String::substr(String::size_type from, String::size_type to) const
     Must(to > from);
 
     String rv;
-    rv.limitInit(rawBuf()+from,to-from);
+    rv.assign(rawBuf()+from,to-from);
```

Comment:
Please fix formatting when changing this line:

```suggestion
    rv.assign(rawBuf()+from, to-from);
```